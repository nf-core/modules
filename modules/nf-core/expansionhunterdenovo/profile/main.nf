process EXPANSIONHUNTERDENOVO_PROFILE {
    tag "$meta.id"
    label 'process_single'

    conda "bioconda::expansionhunterdenovo=0.9.0"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/expansionhunterdenovo:0.9.0--hdc99072_3':
        'quay.io/biocontainers/expansionhunterdenovo:0.9.0--hdc99072_3' }"

    input:
    tuple val(meta), path(alignment_file), path(alignment_index)
    path(fasta)
    path(fasta_fai)

    output:
    tuple val(meta), path("*.locus.tsv")        , emit: locus_tsv
    tuple val(meta), path("*.motif.tsv")        , emit: motif_tsv
    tuple val(meta), path("*.str_profile.json") , emit: str_profile
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    ExpansionHunterDenovo profile \\
        --reads ${alignment_file} \\
        --reference ${fasta} \\
        --output-prefix ${prefix} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        expansionhunterdenovo: \$(echo \$(ExpansionHunterDenovo --help 2>&1) | sed -e "s/ExpansionHunter Denovo v//;s/ Usage.*//")
    END_VERSIONS
    """
}
