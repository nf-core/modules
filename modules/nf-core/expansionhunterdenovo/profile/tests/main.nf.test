
nextflow_process {

    name "Test Process EXPANSIONHUNTERDENOVO_PROFILE"
    script "../main.nf"
    process "EXPANSIONHUNTERDENOVO_PROFILE"

    tag "modules"
    tag "modules_nfcore"
    tag "expansionhunterdenovo"
    tag "expansionhunterdenovo/profile"

    test("test-expansionhunterdenovo-profile") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
				]
				input[1] = [ [id:'fasta'],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
				input[2] = [ [id:'fasta_fai'],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true) ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test-expansionhunterdenovo-profile-cram") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test.paired_end.sorted.cram', checkIfExists: true),
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test.paired_end.sorted.cram.crai', checkIfExists: true)
				]
				input[1] = [ [id:'fasta'],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
				input[2] = [ [id:'fasta_fai'],file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true) ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
