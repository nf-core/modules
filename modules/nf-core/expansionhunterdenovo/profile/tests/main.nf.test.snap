{
    "test-expansionhunterdenovo-profile": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.locus.tsv:md5,225b2e8975f3ba50058db999824fef66"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.motif.tsv:md5,7828d4d4c33ad4a8d70313f9e89758d7"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.str_profile.json:md5,8709992cefee025118798df3ca878d0d"
                    ]
                ],
                "3": [
                    "versions.yml:md5,db98daacc305d0cba3ff6103a62a1677"
                ],
                "locus_tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.locus.tsv:md5,225b2e8975f3ba50058db999824fef66"
                    ]
                ],
                "motif_tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.motif.tsv:md5,7828d4d4c33ad4a8d70313f9e89758d7"
                    ]
                ],
                "str_profile": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.str_profile.json:md5,8709992cefee025118798df3ca878d0d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,db98daacc305d0cba3ff6103a62a1677"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-28T13:35:00.483786"
    },
    "test-expansionhunterdenovo-profile-cram": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.locus.tsv:md5,225b2e8975f3ba50058db999824fef66"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.motif.tsv:md5,7828d4d4c33ad4a8d70313f9e89758d7"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.str_profile.json:md5,8709992cefee025118798df3ca878d0d"
                    ]
                ],
                "3": [
                    "versions.yml:md5,db98daacc305d0cba3ff6103a62a1677"
                ],
                "locus_tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.locus.tsv:md5,225b2e8975f3ba50058db999824fef66"
                    ]
                ],
                "motif_tsv": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.motif.tsv:md5,7828d4d4c33ad4a8d70313f9e89758d7"
                    ]
                ],
                "str_profile": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.str_profile.json:md5,8709992cefee025118798df3ca878d0d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,db98daacc305d0cba3ff6103a62a1677"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-28T13:35:06.781505"
    }
}